oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/config/NamingControl.java

```diff
@@ -95,6 +95,10 @@ public String formatInstanceName(String instanceName) {
      * @return the string, which length less than or equals {@link #endpointNameMaxLength};
      */
     public String formatEndpointName(String serviceName, String endpointName) {
+        if (endpointName == null) {
+            return null;
+        }
```

Comment:
I think you forget the null/empty check for service Name.

Reply:
OK, I add the `service name` checking logic.